lib/components/map/elevation-point-marker.tsx

```diff
@@ -0,0 +1,64 @@
+import { connect } from 'react-redux'
+import { Leg } from '@opentripplanner/types'
+import { Marker } from 'react-map-gl'
+import coreUtils from '@opentripplanner/core-utils'
+import React from 'react'
+import styled from 'styled-components'
+
+type Props = {
+  diagramLeg: Leg
+  elevationPoint: number
+  showElevationProfile?: boolean
+}
+
+/**
+ * As the OTP user moves the cursor over the elevation tracking chart
+ * of a walking or biking leg (to see which point of their itinerary is at which elevation),
+ * ElevationPointMarker displays and moves a marker on the map to highlight
+ * the location that corresponds to the cursor position on the elevation chart,
+ * so the user can see the streets and paths that correspond to a portion of an elevation profile.
+ */
+const ElevationPointMarker = (props: Props) => {
+  const { diagramLeg, elevationPoint, showElevationProfile } = props
+
+  const ElevationMarker = styled.div`
+    background-color: #87cefa;
+    border: 2px solid #fff;
+    border-radius: 50%;
+    height: 15px;
+    width: 15px;
+  `
+
+  // Compute the elevation point marker, if activeLeg and elevation profile is enabled.
+  let elevationPointMarker = null
```

Comment:
Could you just return this variable inside the ifs and return null at the end of the function to eliminate this `let`?

Reply:
This code is marked as new because it was typescripted, but this is all inherited code. I'm not sure about the edge cases that we need to maintain here, so am weary of making changes to it in this PR. We could add a new issue to do this in a new PR?